File: Andromeda.AvaloniaApp/Converter/FloatToStringConverter.cs
using Avalonia;
using Avalonia.Controls;
using Avalonia.Data.Converters;
using Avalonia.Media.Imaging;
using Microsoft.FSharp.Core;
using System;
using System.Globalization;

namespace Andromeda.AvaloniaApp.Converter {
    public class FloatToStringConverter : IValueConverter {
        public object Convert(object value, Type targetType,
            object parameter, CultureInfo culture) {
            if (value != null) {
                // Unwrap optiontype if necessary
                if (value is FSharpOption<float>) {
                    value = (value as FSharpOption<float>).Value;
                }
                return string.Format("{0:0.#}", value);
            }
            else {
                return null;
            }
        }

        public object ConvertBack(object value, Type targetType,
            object parameter, CultureInfo culture) {
            return 0;
        }
    }
}

File: Andromeda.AvaloniaApp.FSharp/Converter/FloatToStringConverter.fs
namespace Andromeda.AvaloniaApp.FSharp.Converter

open Avalonia.Data.Converters
open Microsoft.FSharp.Core
open System

type FloatToStringConverter () =
    interface IValueConverter with
        member __.Convert(value, _, _, _) =
            match value with
            | null -> null
            | value ->
                // Unwrap optiontype if necessary
                let value =
                    match value with
                    | :? Option<float> as value ->
                        value.Value
                    | :? float as value->
                        value
                    | _ -> invalidArg "value" "no float or float option"
                String.Format("{0:0.#}", value) :> obj

        member __.ConvertBack(_, _, _, _) =
            invalidOp ""
